sdk/tests/conformance/context/context-creation-reclaims-vram.html

```diff
@@ -0,0 +1,85 @@
+<!--
+Copyright (c) 2019 The Khronos Group Inc.
+Use of this source code is governed by an MIT-style license that can be
+found in the LICENSE.txt file.
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Test that contexts are freed and garbage collected reasonably</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"> </script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description();
+
+const CANVAS_COUNT = 100;
+const SIZE_REQUEST = 23170; // Chrome doesn't even try for >2GB backbuffers.
+
+debug(`SIZE_REQUEST: ${SIZE_REQUEST*SIZE_REQUEST*4/1024/1024|0} MB`);
+
+const wtu = WebGLTestUtils;
+
+const canvasList = []; // Keeps canvases alive
+
+function newCanvas() {
+  const canvas = document.createElement('canvas');
+  canvas.width = canvas.height = SIZE_REQUEST;
+
+  canvasList.push(canvas);
+  canvas.id = canvasList.length;
+
+  canvas.addEventListener("webglcontextlost", function(e) {
+    //e.preventDefault(); // preventDefault enables context restore.
+    debug(`(webglcontextlost on canvas ${canvas.id})`);
+  });
+
+  return canvas;
+}
+
+function testNextContext(minSize) {
+  const canvas = newCanvas();
+  const gl = wtu.create3DContext(canvas, {
+    depth: false,
+    antialias: false,
+  });
+  assertMsg(gl, `Creating context ${canvas.id} should succeed.`);
+  if (gl) {
+    const width = gl.drawingBufferWidth;
+    const height = gl.drawingBufferHeight;
+    debug(`   allocated size: ${width*height*4/1024/1024|0} MB`);
+    assertMsg(width >= minSize,
+              `   drawingBufferWidth (${width}) should be >= ${minSize}`);
+    assertMsg(height >= minSize,
+              `   drawingBufferHeight (${height}) should be >= ${minSize}`);
```

Comment:
Chromium's WebGL implementation handles the scenario of users allocating many WebGL contexts with large back buffers by scaling down the size of newly created back buffers, rather than losing older contexts. We've tried in the past to lift the global limits on the amount of back buffer VRAM that can be allocated, but have never succeeded; crashes have been seen inside OpenGL drivers on various platforms, instead of GL_OUT_OF_MEMORY being reported. I'm not sure that requiring that newly-allocated contexts have the same back buffer size as the originally-allocated context is an invariant we should enforce; let's discuss in real time on tomorrow's working group conference call.